renderdoc/os/posix/android/android_stringio.cpp

```diff
@@ -68,7 +63,15 @@ rdcstr GetTempRootPath()
   // This is the same as returned by getExternalFilesDir(). It might possibly change in the future.
   rdcstr package;
   GetExecutableFilename(package);
-  return "/sdcard/Android/data/" + package + "/files";
+
+  char platformVersionChar[3];
+  __system_property_get("ro.build.version.sdk", platformVersionChar);
```

Comment:
I'm concerned about the possibility of this overwriting the very small string buffer here, either in a future android version or on an older/customised version of android where the contents aren't a two digit number.

Is there a version of this function that can be used first to enumerate the length of the returned string before returning the string itself? That would definitely be preferred. If not then since this function isn't on the hot path it'd be safer to allocate say `char[1024]` to be safer.

Reply:
There is a macro defined for the max size of a property returned by this function (`PROP_MAX_SIZE`, 92), and the function is guaranteed not to write more than `PROP_MAX_SIZE` to the `char *` so I went with that!